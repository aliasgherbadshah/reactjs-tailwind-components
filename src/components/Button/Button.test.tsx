import React from 'react';
import { render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import {
  Default,
  Destructive,
  LeadingIcon,
  OnlyIcon,
  Primary,
  Secondary,
  Tertiary,
  TrailingIcon,
} from './Button.stories';

describe('components/Button', () => {
  const onClick = jest.fn();

  const initializeComponent = () => render(<Default onClick={onClick} />);

  it('should render default button', () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });

  it('should render primary button', () => {
    const { container } = render(<Primary />);
    expect(container).toMatchSnapshot();
  });

  it('should render secondary button', () => {
    const { container } = render(<Secondary />);
    expect(container).toMatchSnapshot();
  });

  it('should render tertiary button', () => {
    const { container } = render(<Tertiary />);
    expect(container).toMatchSnapshot();
  });

  it('should render destructive button', () => {
    const { container } = render(<Destructive />);
    expect(container).toMatchSnapshot();
  });

  it('should render button with leading icon', () => {
    const { container } = render(<LeadingIcon />);
    expect(container).toMatchSnapshot();
  });

  it('should render button with trailing icon', () => {
    const { container } = render(<TrailingIcon />);
    expect(container).toMatchSnapshot();
  });

  it('should render button with icon only', () => {
    const { container } = render(<OnlyIcon />);
    expect(container).toMatchSnapshot();
  });

  it('should call onClick on button click', async () => {
    const { getByTestId } = initializeComponent();

    const button = getByTestId('button') as HTMLInputElement;
    userEvent.click(button);

    await waitFor(() => {
      expect(onClick).toHaveBeenCalledTimes(1);
    });
  });
});
