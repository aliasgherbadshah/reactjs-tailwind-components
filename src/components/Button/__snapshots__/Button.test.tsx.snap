// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button should render button with icon only 1`] = `
<div>
  <button
    class="sc-gsDKAQ inline-flex justify-between items-center border font-medium rounded-md shadow-sm focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent text-structural-0 bg-primary-600 hover:bg-primary-700 focus:ring-primary-500 px-4 py-2 text-sm false"
    data-testid="button"
  >
    <div
      class="flex"
    />
    <div
      class="flex"
    />
    <div
      class="flex"
    />
  </button>
</div>
`;

exports[`components/Button should render button with leading icon 1`] = `
<div>
  <button
    class="sc-gsDKAQ inline-flex justify-between items-center border font-medium rounded-md shadow-sm focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent text-structural-0 bg-primary-600 hover:bg-primary-700 focus:ring-primary-500 px-4 py-2 text-sm false"
    data-testid="button"
  >
    <div
      class="flex"
    />
    <div
      class="flex"
    >
      Click me
    </div>
    <div
      class="flex"
    />
  </button>
</div>
`;

exports[`components/Button should render button with trailing icon 1`] = `
<div>
  <button
    class="sc-gsDKAQ inline-flex justify-between items-center border font-medium rounded-md shadow-sm focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent text-structural-0 bg-primary-600 hover:bg-primary-700 focus:ring-primary-500 px-4 py-2 text-sm false"
    data-testid="button"
  >
    <div
      class="flex"
    />
    <div
      class="flex"
    >
      Click me
    </div>
    <div
      class="flex"
    />
  </button>
</div>
`;

exports[`components/Button should render default button 1`] = `
<div>
  <button
    class="sc-gsDKAQ inline-flex justify-between items-center border font-medium rounded-md shadow-sm focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent text-structural-0 bg-primary-600 hover:bg-primary-700 focus:ring-primary-500 px-4 py-2 text-sm false"
    data-testid="button"
  >
    <div
      class="flex"
    />
    <div
      class="flex"
    >
      Click me
    </div>
    <div
      class="flex"
    />
  </button>
</div>
`;

exports[`components/Button should render destructive button 1`] = `
<div>
  <button
    class="sc-gsDKAQ inline-flex justify-between items-center border font-medium rounded-md shadow-sm focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent text-structural-0 bg-primary-600 hover:bg-primary-700 focus:ring-primary-500 px-4 py-2 text-sm false"
    data-testid="button"
  >
    <div
      class="flex"
    />
    <div
      class="flex"
    >
      Click me
    </div>
    <div
      class="flex"
    />
  </button>
</div>
`;

exports[`components/Button should render primary button 1`] = `
<div>
  <button
    class="sc-gsDKAQ inline-flex justify-between items-center border font-medium rounded-md shadow-sm focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent text-structural-0 bg-primary-600 hover:bg-primary-700 focus:ring-primary-500 px-4 py-2 text-sm false"
    data-testid="button"
  >
    <div
      class="flex"
    />
    <div
      class="flex"
    >
      Click me
    </div>
    <div
      class="flex"
    />
  </button>
</div>
`;

exports[`components/Button should render secondary button 1`] = `
<div>
  <button
    class="sc-gsDKAQ inline-flex justify-between items-center border font-medium rounded-md shadow-sm focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent text-structural-0 bg-primary-600 hover:bg-primary-700 focus:ring-primary-500 px-4 py-2 text-sm false"
    data-testid="button"
  >
    <div
      class="flex"
    />
    <div
      class="flex"
    >
      Click me
    </div>
    <div
      class="flex"
    />
  </button>
</div>
`;

exports[`components/Button should render tertiary button 1`] = `
<div>
  <button
    class="sc-gsDKAQ inline-flex justify-between items-center border font-medium rounded-md shadow-sm focus:outline-none focus:ring-2 focus:ring-offset-2 border-transparent text-structural-0 bg-primary-600 hover:bg-primary-700 focus:ring-primary-500 px-4 py-2 text-sm false"
    data-testid="button"
  >
    <div
      class="flex"
    />
    <div
      class="flex"
    >
      Click me
    </div>
    <div
      class="flex"
    />
  </button>
</div>
`;
